### ISSUE TITLE
"A mutation operation was attempted on a database that did not allow mutations" error on login

### ISSUE THREAD

**Issue by ghost** [element-hq/hydrogen-web#353]
**Describe the bug**
halogen.city as homeserver
returned this error:`Something went wrong: POST https://www.halogen.city/_matrix/client/r0/login?&_cacheBuster=6963521230175225: NetworkError when attempting to fetch resource..`
I did enter my login info.
Meanwhile on another device, i find "Hydrogen" as a Session on my account settings in "Security & Privacy".

**To Reproduce**
1. Go to the main page (where you login), enter login info
2. Click on login
[x] 3. Scroll down to No where
4. See error `Something went wrong: POST https://www.halogen.city/_matrix/client/r0/login?&_cacheBuster=6963521230175225: NetworkError when attempting to fetch resource..`

**Expected behavior**
Access my account

**Screenshots**
![image](https://user-images.githubusercontent.com/16048625/118364647-d2b21d00-b588-11eb-94cd-160e77126033.png)

**Desktop (please complete the following information):**
 - OS: Debian: GNU/Linux 10 (buster) x86_64
 - Browser: Tor browser
 - Version: 10.0.16

**Additional context**
I was using Whonix.

**Comment by bwindels** [comment]
if you go back and try again, does it now work?

**Comment by ghost** [comment]
Same here. Still does not work.
![_](https://user-images.githubusercontent.com/66488718/118518136-1f882600-b738-11eb-8c6d-a97b44ae2211.png)

**Comment by ghost** [comment]
> if you go back and try again, does it now work?

Nope, same output.

**Comment by ghost** [comment]
Ok. I noticed something new if i try "halogen.city" it returns `Something went wrong: IDBRequest failed on <unknown db>.<unknown store>: (name: InvalidStateError) (code: 11) A mutation operation was attempted on a database that did not allow mutations..` and adds a session to my account.
![118534216-e43e2500-b740-11eb-81d9-1bfc5fbecb55](https://user-images.githubusercontent.com/16048625/118566738-5bd57980-b76c-11eb-938b-fb78da12ac0b.png)


But if you try "www.halogen.city" `Something went wrong: POST https://www.halogen.city/_matrix/client/r0/login?&_cacheBuster=1817413715078564: NetworkError when attempting to fetch resource..` and **Does not** add a session to my account.
![118534475-2ff0ce80-b741-11eb-990d-2584cc63b6e5](https://user-images.githubusercontent.com/16048625/118566769-6a239580-b76c-11eb-8e5b-f3a249f064b9.png)

**Comment by ghost** [comment]
@Jonahgay I have now tried several possibilities in the homeserver entry, and came to the conclusion that you have to enter the full URL. "johrpan.de" (my homeserver) did not work in this case, but "https://matrix.johrpan.de" did. Maybe you try, to enter the full URL instead the domain only?

**Comment by ghost** [comment]
@ODIRFY Like this ?
Cause its not doing what it's expected.
![image](https://user-images.githubusercontent.com/16048625/118566819-81fb1980-b76c-11eb-8ced-15b2b57844d0.png)

**Comment by ghost** [comment]
> @ODIRFY Like this ?
> Cause its not doing what it's expected.
> ![image](https://user-images.githubusercontent.com/16048625/118566819-81fb1980-b76c-11eb-8ced-15b2b57844d0.png)

@Jonahgay I already said: "https://matrix.johrpan.de" works for me, the client starts. You have to ask your server admin, where he did host the matrix server (not the element-web instance!) and try to log-in again. But you can easily find that out yourself:
1. Open Element-Web
2. Find an image sent in a room
3. Right mouse click –> View picture in new tab
4. Then you see where the image is located on halogen.city –> Domain of Matrix-Server. 
![_](https://user-images.githubusercontent.com/66488718/118606475-bfd15f80-b7b7-11eb-9c6f-0170ce65db1f.png)

**Comment by ghost** [comment]
None of the solutions here work for me... I have never been able to log in successfully using Hydrogen on any homeserver, even with the latest master branch as of right now. Depending on which homeserver I choose (regardless of whether I prepend `https://` or not), I either get this NetworkError as described above, or "The operation is insecure".

**Comment by bwindels** [comment]
@Jonahgay your homeserver seems to be `halogen.city`, not `www.halogen.city`. Does the former not work for you?

**Comment by bwindels** [comment]
> None of the solutions here work for me... I have never been able to log in successfully using Hydrogen on any homeserver, even with the latest master branch as of right now. Depending on which homeserver I choose (regardless of whether I prepend `https://` or not), I either get this NetworkError as described above, or "The operation is insecure".

Can you please file a separate issue? Mention if you are hosting your own deployment of hydrogen and what browser and homeserver you are using.

**Comment by bwindels** [comment]
> @Jonahgay I have now tried several possibilities in the homeserver entry, and came to the conclusion that you have to enter the full URL. "johrpan.de" (my homeserver) did not work in this case, but "https://matrix.johrpan.de" did. Maybe you try, to enter the full URL instead the domain only?

This is because Hydrogen does not support .well-known lookups yet, I've created https://github.com/vector-im/hydrogen-web/issues/379 to track this.

**Comment by ghost** [comment]
> @Jonahgay your homeserver seems to be `halogen.city`, not `www.halogen.city`. Does the former not work for you?

I did try it, and mention it in this post:
https://github.com/vector-im/hydrogen-web/issues/353#issuecomment-842518457

Tldr it returned : `Something went wrong: IDBRequest failed on <unknown db>.<unknown store>: (name: InvalidStateError) (code: 11) A mutation operation was attempted on a database that did not allow mutations..` and created a session on my account.

**Comment by bwindels** [comment]
@Jonahgay  You're not in a private window or anything like that, are you?

If not, you can extract logs by opening the developer tools and entering this in the console:
```js
__hydrogenViewModel.logger.export().then(export => export.asBlob()).readAsBuffer().then(buffer => new TextDecoder().decode(buffer))
```
Which will give you the debug logs [1] as a big blob of JSON. If you wouldn't mind passing me that as a file, it would help me determine where exactly things go wrong. You can contact me on matrix, I'm @bwindels:matrix.org.

1: Debug logs contain application usage data including your username, the IDs or aliases of the rooms or groups you have visited, the usernames of other users and the names of files you send. They do not contain messages.

**Comment by pratikbin** [comment]
I'm getting :arrow_down: . FYI, Federation is off.

![image](https://user-images.githubusercontent.com/68642400/152929348-81e974b6-5797-4580-9f9e-dcedd8ca6621.png)

**Comment by bwindels** [comment]
Looks like the idb error is #797 (user never confirmed it was a private window or not, but seems like most likely explanation)
